api/agent/slots.go

```diff
@@ -326,16 +326,11 @@ func getSlotQueueKey(call *call) string {
 	}
 
 	var buf [sha1.Size]byte
-	byts := hash.Sum(buf[:0])
-	return unsafeString(byts)
+	hash.Sum(buf[:0])
```

Comment:
shouldn't this be something like this?

```
return string(hash.Sum(nil))
```

Reply:
that is effectively the same, since go's escape analysis stinks. the contract of  `hash.Hash.Sum([]byte)` is that it will append the result to the provided byte slice, returning the resulting slice. in theory, providing your own perfectly sized byte slice should avoid an allocation, in practice it seems to simply allow one to re-use a byte slice to do repeated sums instead of allocating a new one each sum.

Reply:
example: https://play.golang.org/p/Ry4G5N2KDwP

Reply:
OK, got confused by sha1 docs, different function:
https://golang.org/pkg/crypto/sha1/#Sum

Reply:
we should prob use another hash function like siphash / xxhash but it's an extra import. those tend to provide a more allocation friendly api than `hash.Hash` and we don't need a cryptographic hash function for this... once we add versioning of routes / apps this dies tho.